### PR TITLE
8286360: ARM32: Fix crashes after JDK-8284161 (Virtual Threads)

### PR DESCRIPTION
**Pull request by shipilev**
Virtual Threads are not fully implemented on ARM32, but there is a fallback implementation that can be used. Still, there is some VM code that is executed unconditionally anyway, crashing the VM on some tests. This improves the ARM32 code to the point that VM is able to build itself and pass tests. This adds the scaffolding for full implementation too.

Additional testing:
 - [x] Linux arm32 fastdebug build
 - [x] Linux arm32 fastdebug, `jdk_loom  hotspot_loom` pass
 - [x] Linux arm32 fastdebug, `jdk_jfr` passes (some timeout failures, because the board I have is very slow)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8286360](https://bugs.openjdk.java.net/browse/JDK-8286360): ARM32: Fix crashes after JDK-8284161 (Virtual Threads)

### Reviewers
 * [Dmitry Samersoff](https://openjdk.java.net/census#dsamersoff) (@dsamersoff - **Reviewer**) ⚠️ Review applies to [dff69a64](https://git.openjdk.java.net/jdk/pull/8634/files/dff69a64b09f2596b7f2f350c9e51e22818d6ca7)

### Contributors
 * Sergey Nazarkin `<snazarki@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8634/head:pull/8634` \
`$ git checkout pull/8634`

Update a local copy of the PR: \
`$ git checkout pull/8634` \
`$ git pull https://git.openjdk.java.net/jdk pull/8634/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8634`

View PR using the GUI difftool: \
`$ git pr show -t 8634`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8634.diff">https://git.openjdk.java.net/jdk/pull/8634.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/arm/stubGenerator_arm.cpp [hunk 1]
```diff
@@ -3029,11 +3097,13 @@ class StubGenerator: public StubCodeGenerator {
...
-    if (all) {
-      generate_all();
-    } else {
+  StubGenerator(CodeBuffer* code, int phase) : StubCodeGenerator(code) {
+    if (phase == 0) {
       generate_initial();
+    } else if (phase == 1) {
+      generate_phase1();
+    } else {
+      generate_all();
```
Comment (dsamersoff) [inline src/hotspot/cpu/arm/stubGenerator_arm.cpp:3106]: It might be better to check for phases explicitly:
  
     } else if (phase == 2) {
       generate_all();
     } else {
        ShouldNotReachHere();
     }

Comment (shipilev) [reply src/hotspot/cpu/arm/stubGenerator_arm.cpp:3106]: I agree, but it matches what all other ports do :)

File: src/hotspot/share/runtime/arguments.cpp [hunk 2]
```diff
@@ -4205,6 +4205,7 @@ jint Arguments::apply_ergo() {
   }
 #endif // COMPILER2
 
+
```
Comment (dsamersoff) [inline src/hotspot/share/runtime/arguments.cpp:4208]: Unnecessary changes, that bring one more file to diff list. 
Please fix.

File: src/hotspot/cpu/arm/frame_arm.inline.hpp [hunk 3]
```diff
@@ -40,16 +40,17 @@ inline frame::frame() {
   _cb = NULL;
   _deopt_state = unknown;
   _on_heap = false;
+  _oop_map = NULL;
   DEBUG_ONLY(_frame_index = -1;)
 }
 
 inline frame::frame(intptr_t* sp) {
```
Comment (dsamersoff) [inline src/hotspot/cpu/arm/frame_arm.inline.hpp:47]: Will AArch64 code work?

inline frame::frame(intptr_t* sp) : frame(sp, sp, *(intptr_t**)(sp - frame::sender_sp_offset), *(address*)(sp - 1)) {}

Comment (dsamersoff) [reply src/hotspot/cpu/arm/frame_arm.inline.hpp:47]: Nice to have comments here to avoid spending time on it in the future, but it's not crucial.

File: src/hotspot/share/runtime/arguments.cpp (hunk 2 above)
Comment (shipilev) [reply src/hotspot/share/runtime/arguments.cpp]: Fixed.

**Review by dsamersoff — APPROVED** [review]
